packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts

```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
         },
       ],
     },
+    {
+      // https://github.com/typescript-eslint/typescript-eslint/issues/2775
+      code: `
+        import Default, * as Rest from 'module';
+        const a: Rest.A = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type * as Rest from 'module';
+import Default from 'module';
+        const a: Rest.A = '';
+      `,
+      errors: [
+        {
+          messageId: 'aImportIsOnlyTypes',
+          line: 2,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+        import Default, * as Rest from 'module';
+        const a: Default = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type Default from 'module';
+import  * as Rest from 'module';
```

Comment:
Hard to notice, but this is being fixed with a double space: `import  *`.
We probably don't want to be doing that if we can avoid it.